code/modules/plumbing/plumbers/pill_press.dm

```diff
@@ -1,111 +1,116 @@
-	///maximum size of a bottle
-	var/max_bottle_volume = 50
-	///current operating product (pills or patches)
+	/// current operating product (pills or patches)
 	var/product = "pill"
-	///the minimum size a pill or patch can be
-	var/min_volume = 5
-	///the maximum size a pill or patch can be
-	var/max_volume = 50
-	///selected size of the product
+	/// selected size of the product
 	var/current_volume = 10
-	///prefix for the product name
+	/// prefix for the product name
 	var/product_name = "factory"
-	///the icon_state number for the pill.
-	var/pill_number = RANDOM_PILL_STYLE
-	///list of id's and icons for the pill selection of the ui
-	var/list/pill_styles
-	/// Currently selected patch style
-	var/patch_style = DEFAULT_PATCH_STYLE
-	/// List of available patch styles for UI
-	var/list/patch_styles
-	///list of products stored in the machine, so we dont have 610 pills on one tile
+	/// All packaging types wrapped up in 1 big list
+	var/static/list/packaging_types = null
+	///The type of packaging to use
+	var/packaging_type
+	///Category of packaging
+	var/packaging_category
+	/// list of products stored in the machine, so we dont have 610 pills on one tile
 	var/list/stored_products = list()
-	///max amount of pills allowed on our tile before we start storing them instead
-	var/max_floor_products = 10
-
-/obj/machinery/plumbing/pill_press/examine(mob/user)
-	. = ..()
-	. += span_notice("The [name] currently has [stored_products.len] stored. There needs to be less than [max_floor_products] on the floor to continue dispensing.")
 
 /obj/machinery/plumbing/pill_press/Initialize(mapload, bolt, layer)
 	. = ..()
 
+	if(!packaging_types)
+		var/datum/asset/spritesheet/simple/assets = get_asset_datum(/datum/asset/spritesheet/chemmaster)
+
+		var/list/types = list(
+			CAT_PILLS = GLOB.reagent_containers[CAT_PILLS],
+			CAT_PATCHES = GLOB.reagent_containers[CAT_PATCHES],
+			"Bottles" = list(/obj/item/reagent_containers/cup/bottle),
+		)
+
+		packaging_types = list()
+		for(var/category in types)
+			var/list/packages = types[category]
+
+			var/list/category_item = list("cat_name" = category)
+			for(var/obj/item/reagent_containers/container as anything in packages)
+				var/list/package_item = list(
+					"class_name" = assets.icon_class_name(sanitize_css_class_name("[container]")),
+					"ref" = REF(container)
+				)
+				category_item["products"] += list(package_item)
+
+			packaging_types += list(category_item)
+
+	packaging_type = REF(GLOB.reagent_containers[CAT_PILLS][1])
+	_decode_category()
+
 	AddComponent(/datum/component/plumbing/simple_demand, bolt, layer)
 
+/obj/machinery/plumbing/pill_press/examine(mob/user)
+	. = ..()
+	. += span_notice("The [name] currently has [stored_products.len] stored. There needs to be less than [MAX_FLOOR_PRODUCTS] on the floor to continue dispensing.")
+
+/// decode product category from it's type path and returns the decoded typepath
+/obj/machinery/plumbing/pill_press/proc/_decode_category()
```

Comment:
```suggestion
/obj/machinery/plumbing/pill_press/proc/decode_category()
```

Reply:
I saw somewhere that if you want to create an internal private proc which is not meant to be used anywhere outside then you start it's name with underscore.

For example _SendSignal() that's what I was going for. But yeah I guess that's not exactly needed here